core/dbt/deps/resolver.py

```diff
@@ -116,21 +116,21 @@ def _check_for_duplicate_project_names(
 
 
 def resolve_packages(
-    packages: List[PackageContract], config: RuntimeConfig
+    packages: List[PackageContract], project: Project, profile, cli_vars
 ) -> List[PinnedPackage]:
     pending = PackageListing.from_contracts(packages)
     final = PackageListing()
-    renderer = DbtProjectYamlRenderer(config, config.cli_vars)
+    renderer = DbtProjectYamlRenderer(profile, cli_vars)
```

Comment:
If you don't need to use Target for resolving the package, you don't need to initialize this render with `profile`.

Reply:
Given that any dbt_project.yml (including those from packages) _can_ include Jinja expressions that use {{ target }}, it feels necessary to continue to pass `UnsetProfile` through the deps command to stay consistent with the current behaviour and not raise `CompilationException: 'target' is undefined` errors

Reply:
But shouldn't we already rendered everything at the time we start running the task?